app/components/Modals/VotingModal/VotingModal.jsx

```diff
@@ -0,0 +1,246 @@
+                  {textStrings.CANDIDATES}
+                </div>
+                <div className={classNames(styles.rightCol, styles.title)}>
+                  {textStrings.NUMBER_OF_VOTES}
+                </div>
+              </div>
+              {this.renderNodeList()}
+            </div>
+          </div>
+          <Button
+            className={styles.submitButton}
+            onClick={this.handleCastVote}
+            disabled={!this.hasVotes()}
+          >{textStrings.CAST_VOTE}</Button>
+        </div>
+        {this.renderInformationModal()}
+        {this.renderVoteConfirmationModal()}
+      </BaseModal>
+    )
+  }
+
+  renderNodeList () {
+    return placehoderData.map((data, index) => {
+      const {address, votes} = data
+      return (
+        <div key={index} className={styles.row}>
+          <div className={styles.leftCol}>
+            <input className={styles.checkBox} type='checkbox' onClick={() => this.handleNodeSelected(index)} />
+            <a href=''>{address}</a>
+          </div>
+          <div className={styles.rightCol}>
+            {votes}
+          </div>
+        </div>
+      )
+    })
+  }
+
+  handleCastVote () {
+    this.hasVotes() && this.showConfirmationModal()
+  }
+
+  hasVotes () {
+    const {votes} = this.state
+    return Object.keys(votes).reduce((result, key) => result || !!votes[key], false)
+  }
+
+  handleNodeSelected (index: number) {
+    const {votes} = this.state
+    this.setState({votes: {...votes, [index.toString()]: !votes[index.toString()]}})
+  }
+
+  showInfoModal () {
+    this.setState({infoShowing: true})
+  }
+
+  hideInfoModal () {
+    this.setState({infoShowing: false})
+  }
+
+  showConfirmationModal () {
+    this.setState({confirmationShowing: true})
+  }
+
+  hideConfirmationModal () {
+    this.setState({confirmationShowing: false})
+  }
+
+  renderInformationModal () {
+    const { infoShowing } = this.state
+
+    return infoShowing && (
+      <BaseModal
+        title={textStrings.EXPLANATION_TITLE}
+        hideModal={this.hideInfoModal}
+        style={{
+          content: {
+            width: '43rem',
+            height: '15rem'
+          }
+        }}
+      >
+        {textStrings.VOTING_EXPLANATION}
+      </BaseModal>
+    )
+  }
+
+  renderVoteConfirmationModal () {
+    const { confirmationShowing, votes } = this.state
+
+    return confirmationShowing && (
+      <BaseModal
+        title={textStrings.CONFIRMATION_TITLE}
+        hideModal={this.hideConfirmationModal}
+        style={{
+          content: {
+            width: '43rem',
+            height: '25rem'
+          }
+        }}
+      >
+        {textStrings.CONFIRMATION_MESSAGE.replace('<number_of_votes>', votesAvailable.toString())}
+        <ul className={styles.addressList}>
+          {Object.keys(votes).reduce((elements, key, index) => {
+            if (votes[key]) {
+              elements.push(
+                <li key={index}>{placehoderData[Number(key)].address}</li>
```

Comment:
Could you use a more unique key? Its best when its actual ids, but if you use index then - perhaps `voteRow${index}`